### PR TITLE
docs: add TigerBeetle in Your System Architecture page

### PR DESCRIPTION
**Pull request by emschwartz**

### CODE REVIEW

File: docs/develop/tigerbeetle-in-your-architecture.md
Comment (sentientwaffle) [inline docs/develop/tigerbeetle-in-your-architecture.md]:
```suggestion
integers. Most likely, you will want these integers to map to enums of type names or strings, along
with other associated metadata.
```

**Comment by jh125486** [comment]
Is there a specific reason for using the term "OLGP"? 
I'm not familiar with it as a standard term in the industry/academia. Could someone clarify? 🤷

**Comment by mankins** [comment]
@jh125486 GP = General Purpose, and as I understand it the architectural pattern that TigerBeetle is going after is to combine together a group of databases each with their own purpose. GP data is stuff that doesn't change on a per-transaction basis...things like account owner's address, etc. 

I think some of the challenges with documenting TigerBeetle are related to this concept of "it doesn't store everything" and so perhaps the documenter wanted to make this more clear with the different database types. 

FWIW, I wasn't familiar with that term either, but it made sense in context and helped me understand the particular TigerBeetle POV.

**Comment by matklad** [comment]
>I'm not familiar with it as a standard term in the industry/academia. Could someone clarify? 🤷

It is not a standard term _yet_, but we are trying to popularize it, because the big idea behind TigerBeetle is to tease apart general purpose processing (where the shape of your data evolves over time, and where the data and operations on it are defined in terms of some flexible query language a la SQL) and transaction processing (where the shape of your data is mostly fixed, the data is dense, supports only a closed set of operations, but where the operations are very very fast). 

But it would probably help to include "Did you just made up OLGP term?" entry to FAQ or something!

File: docs/develop/tigerbeetle-in-your-architecture.md
Comment (sentientwaffle) [inline docs/develop/tigerbeetle-in-your-architecture.md]: Maybe mention strict serializability and/or link  to `docs/develop/consistency.md`?

File: docs/develop/tigerbeetle-in-your-architecture.md
Comment (sentientwaffle) [inline docs/develop/tigerbeetle-in-your-architecture.md:1]: `system-architecture.md`/`architecture.md`? (I like short paths :wink:).

**Comment by jh125486** [comment]
> @jh125486 GP = General Purpose, and as I understand it the architectural pattern that TigerBeetle is going after is to combine together a group of databases each with their own purpose. GP data is stuff that doesn't change on a per-transaction basis...things like account owner's address, etc.
> 
> I think some of the challenges with documenting TigerBeetle are related to this concept of "it doesn't store everything" and so perhaps the documenter wanted to make this more clear with the different database types.
> 
> FWIW, I wasn't familiar with that term either, but it made sense in context and helped me understand the particular TigerBeetle POV.

Possibly just calling it a `SQL/NoSQL DB` or  `general purpose database` might be the simplest? 

Even prefixing it with "OL" is sorta weird... I don't think TigerBeetle is targeting the elusive "underground bunker tape database market" :)

**Comment by jorangreef** [comment]
@jh125486 here's our thinking (and rationale) on teasing general purpose workloads out of the transaction processing workload (in much the same way as OLAP and streaming and blob storage were teased out of OLTP):

https://www.infoq.com/presentations/redesign-oltp/

File: docs/develop/system-architecture.md
Comment (sentientwaffle) [inline docs/develop/system-architecture.md]: Do we deliberately avoid referring to these as "data plane" / "control plane"?

File: docs/assets/tigerbeetle-in-your-architecture.png
Comment (sentientwaffle) [inline docs/assets/tigerbeetle-in-your-architecture.png:1]: `tidy` is failing in CI due to:

```
2024-04-18T17:12:24.3169750Z run test: error: 'test.tidy no large blobs' failed: blob 302577 docs/assets/tigerbeetle-in-your-architecture.png
2024-04-18T17:12:24.3172900Z /Users/runner/work/tigerbeetle/tigerbeetle/src/tidy.zig:250:26: 0x10ce53931 in test.tidy no large blobs (test)
2024-04-18T17:12:24.3176380Z     if (has_large_blobs) return error.HasLargeBlobs;
2024-04-18T17:12:24.3177820Z                          ^
2024-04-18T17:12:24.3179960Z run test: error: 'test.tidy extensions' failed: bad extension: docs/assets/tigerbeetle-in-your-architecture.png
2024-04-18T17:12:24.3182740Z /Users/runner/work/tigerbeetle/tigerbeetle/src/tidy.zig:288:24: 0x10ce55826 in test.tidy extensions (test)
2024-04-18T17:12:24.3184740Z     if (bad_extension) return error.BadExtension;
2024-04-18T17:12:24.3185860Z                        ^
2024-04-18T17:12:24.3186550Z run test: error: ^
```

We could whitelist the diagram.
Or, alternatively (since we try to keep large blobs out of the repo) could it be an SVG instead? I expect that would be smaller.

File: docs/develop/data-modeling.md
Comment (sentientwaffle) [inline docs/develop/data-modeling.md]: broken link label

File: docs/assets/tigerbeetle-in-your-architecture.png
Comment (emschwartz) [reply docs/assets/tigerbeetle-in-your-architecture.png:1]: We're going to replace this with an illustration before this is merged. I'll see if we can use an SVG

Comment (sentientwaffle) [reply docs/assets/tigerbeetle-in-your-architecture.png:1]: Ah, if it is an actual illustration then an image might make sense after all (depending on the art style). Either works!

Comment (matklad) [reply docs/assets/tigerbeetle-in-your-architecture.png:1]: Hm, I'd be against adding a 300KiB image to the repo even if this is just this one single time :-) One image won't break us, but it is exceedingly unlikely that we'll stop at one image. 

I think what we should do here is include a link to the image hosted elsewhere. Ideally, on our website, but I know this one weird trick...

Comment (matklad) [reply docs/assets/tigerbeetle-in-your-architecture.png:1]: tigerbeetle-in-your-architecture

Comment (matklad) [reply docs/assets/tigerbeetle-in-your-architecture.png:1]: @emschwartz you can use `https://github.com/tigerbeetle/tigerbeetle/assets/1711539/ffd8d792-ec7f-4211-b84c-0a0451a2c037` as a link to the image above (hosted by GitHub).

Comment (emschwartz) [reply docs/assets/tigerbeetle-in-your-architecture.png:1]: Done! I was also going to wait until the proper illustration is finished to merge this PR, but I think we could just merge it and update it with the illustration later

Comment (matklad) [reply docs/assets/tigerbeetle-in-your-architecture.png:1]: Yeah, I think it's also better to split the work into separately landable chunks, if possible! The fewer things in the air the better!

**Review by matklad — APPROVED** [review]

**Comment by emschwartz** [comment]
We'll update the image with this one:

data_plane12

**Pull request merged**
